holoviews/core/data/xarray.py

```diff
@@ -272,9 +272,9 @@ def length(cls, dataset):
     @classmethod
     def dframe(cls, dataset, dimensions):
         if dimensions:
-            return dataset.reindex(columns=dimensions).data.to_dataframe().reset_index(dimensions)
+            return dataset.data.to_dataframe().reset_index()[dimensions]
         else:
-            return dataset.data.to_dataframe().reset_index(dimensions)
+            return dataset.data.to_dataframe()
```

Comment:
Does this result in the same behavior as before (just written more clearly) or is the index no longer reset in the second branch?

Reply:
Thanks, fixed and added a unit test for that branch too.